components/brave_wallet/browser/solana_tx_manager_unittest.cc

```diff
@@ -1423,4 +1427,83 @@ TEST_F(SolanaTxManagerUnitTest, ProcessSolanaHardwareSignature) {
                                      "");
 }
 
+TEST_F(SolanaTxManagerUnitTest, RebroadcastTransaction) {
+  auto data = GetSystemProgramTransferTxData(sol_account(), to_account());
+  ASSERT_TRUE(data);
+  SetInterceptor(latest_blockhash1_, last_valid_block_height1_, "", "", true,
+                 last_valid_block_height1_, "[null]");
+
+  std::string meta_id1;
+  AddUnapprovedTransaction(mojom::kSolanaMainnet, data.Clone(), sol_account(),
+                           &meta_id1);
+  ASSERT_FALSE(meta_id1.empty());
+
+  // No rebroadcast when first sending out the tx.
+  ApproveTransaction(meta_id1);
+  task_environment_.RunUntilIdle();
+  EXPECT_EQ(send_transaction_calls_, 1u);
+
+  // Rebradcast should happen once for tx with null signature status.
+  send_transaction_calls_ = 0;
+  SetInterceptor(latest_blockhash2_, last_valid_block_height2_, "", "", true,
+                 last_valid_block_height1_, "[null]");
+  task_environment_.FastForwardBy(
+      base::Seconds(kSolanaBlockTrackerTimeInSeconds));
+  task_environment_.RunUntilIdle();
```

Comment:
Is this necessary after `FastForwardBy`?

Reply:
It's for waiting block tracker actions triggered with this forward.

Reply:
Fixed in https://github.com/brave/brave-core/pull/23329/commits/7cc18ad5a5f0ef1ca890d6eb24f5d3da6a50a255